astropy/config/tests/test_configs.py

```diff
@@ -37,26 +37,122 @@ def test_paths():
     assert "testpkg" in paths.get_cache_dir(rootname="testpkg")
 
 
+@pytest.fixture
+def isolated_setup(monkeypatch):
+    # ignore global state of the test session
+    monkeypatch.delenv("ASTROPY_CACHE_DIR", raising=False)
+    monkeypatch.delenv("ASTROPY_CONFIG_DIR", raising=False)
+    monkeypatch.delenv("XDG_CACHE_HOME", raising=False)
+    monkeypatch.delenv("XDG_CONFIG_HOME", raising=False)
+
+    monkeypatch.setattr(paths.set_temp_cache, "_temp_path", None)
+    monkeypatch.setattr(paths.set_temp_config, "_temp_path", None)
+
+
 @pytest.mark.parametrize(
-    "environment_variable,func",
+    "env_var_template",
     [
-        # Regression test for #17514 - XDG_CACHE_HOME had no effect
```

Comment:
This comment should be kept.

Reply:
@eerovaher , so you want a new comment added with different phrasing, still?